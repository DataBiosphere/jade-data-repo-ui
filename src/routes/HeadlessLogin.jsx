import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { logIn } from '../actions/index';
import { withStyles } from '@material-ui/core/styles';

const styles = (theme) => ({
  wrapper: {
    marginTop: 500,
  },
});

export class HeadlessLogin extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      token: '',
    };
  }

  static propTypes = {
    classes: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  handleChange = (event) => {
    this.setState({
      token: event.target.value,
    });
  };

  handleLoginButton = () => {
    const { dispatch } = this.props;
    const { token } = this.state;

    const expireTime = Date.now() + 8640000;

    dispatch(logIn('user.name', 'user.imageUrl', 'user.email', token, expireTime));
  };

  render() {
    const { classes } = this.props;

    return (
      <div className={classes.wrapper}>
        <h1>michael's secret login page</h1>
        <TextField id="tokenInput" label="Password" type="password" onChange={this.handleChange} />
        <Button id="e2eLoginButton" onClick={this.handleLoginButton}>
          Log In
        </Button>
      </div>
    );
  }
}

// eslint-disable-next-line no-unused-vars
function mapStateToProps(state) {
  return {};
}

export default connect(mapStateToProps)(withStyles(styles)(HeadlessLogin));
