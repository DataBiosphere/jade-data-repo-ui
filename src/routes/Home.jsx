import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';

import WelcomeView from '../components/WelcomeView';

const styles = (theme) => ({
  alignItems: 'center',
  display: 'flex',
  flexDirection: 'column',
  fontFamily: theme.typography.fontFamily,
  minHeight: '100vh',
});

export class Home extends React.PureComponent {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  render() {
    const { classes, dispatch } = this.props;

    return (
      <div key="home" data-testid="home" className={classes.styles}>
        <WelcomeView dispatch={dispatch} />
      </div>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return { user: state.user };
}

export default connect(mapStateToProps)(withStyles(styles)(Home));
