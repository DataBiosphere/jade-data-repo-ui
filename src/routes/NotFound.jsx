import React from 'react';
import { Link } from 'react-router-dom';

function NotFound() {
  return (
    <div key="404">
      <h1>404</h1>
      <Link to="/">
        <h2>go home</h2>
      </Link>
    </div>
  );
}

export default NotFound;
