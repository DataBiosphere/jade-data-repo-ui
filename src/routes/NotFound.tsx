import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import OopsISpilledMyTerra from 'media/images/terra404_mug.svg';
import { withStyles, ClassNameMap } from '@mui/styles';
import { Button, Grid, Typography } from '@mui/material';
import { CustomTheme } from '@mui/material/styles';

const styles = (theme: CustomTheme) => ({
  pageRoot: {
    padding: '16px 24px',
    height: '75%',
  },
  h1: {
    ...theme.typography.h1,
    color: theme.typography.color,
    fontSize: '10rem',
    lineHeight: '1',
  },
  h2: {
    ...theme.typography.h1,
    color: theme.typography.color,
  },
  text: {
    ...theme.typography.body1,
    fontSize: '1.25rem',
    color: theme.typography.color,
  },
  button: {
    marginTop: '1em',
  },
  buttonText: {
    color: theme.palette.common.white,
  },
});

function NotFound(props: { classes: ClassNameMap }) {
  const { classes } = props;
  return (
    <Grid
      className={classes.pageRoot}
      container
      spacing={1}
      alignContent="center"
      alignItems="center"
    >
      <Grid style={{ textAlign: 'center' }} item xs={6}>
        <Typography className={classes.h1} variant="h1">
          404
        </Typography>
        <Typography className={classes.h2} variant="h2">
          Page Not Found
        </Typography>
        <Typography className={classes.text}>
          We're sorry, the page you requested could not be found{' '}
        </Typography>
        <Button
          className={classes.button}
          component={RouterLink}
          disableElevation
          color="primary"
          variant="contained"
          size="large"
          to="/"
        >
          <span className={classes.buttonText}>Return to Dashboard</span>
        </Button>
      </Grid>
      <Grid item xs={6}>
        <img alt="404NotFound" src={OopsISpilledMyTerra} />
      </Grid>
    </Grid>
  );
}

export default withStyles(styles)(NotFound);
