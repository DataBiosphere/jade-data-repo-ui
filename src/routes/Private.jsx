import React from 'react';
import { BrowserRouter, Link, Switch, Route } from 'react-router-dom';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

import HomeView from '../components/HomeView';
import StudiesView from '../components/StudyView';
import DatasetView from '../components/DatasetView';
import DatasetCreateView from '../components/DatasetCreateView';
import DatasetPreviewView from '../components/DatasetPreviewView';
import DatasetDetailView from '../components/DatasetDetailView';

const styles = theme => ({
  mainContent: {
    padding: theme.spacing.unit * 3, // TODO this seems to be causing the computedMatch console error
  },
  tabsIndicator: {
    borderBottom: '4px solid #74ae43',
  },
  tabsRoot: {
    paddingLeft: theme.spacing.unit * 3,
    borderBottom: '2px solid #74ae43',
    color: '#5c912e',
  },
  tabRoot: {},
  tabSelected: {
    backgroundColor: '#ddebd0',
  },
});

class Private extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '/',
    };
  }

  static propTypes = {
    classes: PropTypes.object.isRequired,
  };

  handleChange = (event, value) => {
    this.setState({ value });
  };

  render() {
    const { value } = this.state;
    const { classes } = this.props;
    return (
      <BrowserRouter>
        <div>
          <Tabs
            value={value}
            onChange={this.handleChange}
            classes={{ root: classes.tabsRoot, indicator: classes.tabsIndicator }}
          >
            <Tab
              label="Home"
              component={Link}
              value="/"
              to="/"
              classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
            />
            <Tab
              label="Studies"
              component={Link}
              value="/studies"
              to="/studies"
              classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
            />
            <Tab
              label="Datasets"
              component={Link}
              value="/datasets"
              to="/datasets"
              classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
            />
          </Tabs>
          <Switch>
            <div className={classes.mainContent}>
              <Route exact path="/" component={HomeView} />
              <Route exact path="/studies" component={StudiesView} />
              <Route exact path="/datasets" component={DatasetView} />
              <Route exact path="/datasets/create" component={DatasetCreateView} />
              <Route exact path="/datasets/preview" component={DatasetPreviewView} />
              <Route exact path="/dataset/:uuid" component={DatasetDetailView} />
            </div>
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default withStyles(styles)(Private);
