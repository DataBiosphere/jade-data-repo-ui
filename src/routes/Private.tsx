import React, { Fragment } from 'react';
import { Link, Redirect, Route, Router, Switch } from 'react-router-dom';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import { CustomTheme } from '@mui/material/styles';
import { createStyles, WithStyles, withStyles } from '@mui/styles';
import { ConnectedRouter } from 'connected-react-router';
import history from 'modules/hist';

import HelpContainer from '../components/help/HelpContainer';
import HomeView from '../components/HomeView';
import DatasetDataView from '../components/dataset/data/DatasetDataView';
import DatasetOverview from '../components/dataset/overview/DatasetOverview';
import DatasetSchemaCreationView from '../components/dataset/schemaCreation/DatasetSchemaCreationView';
import SnapshotDataView from '../components/snapshot/data/SnapshotDataView';
import SnapshotOverview from '../components/snapshot/overview/SnapshotOverview';
import { TabClasses } from '@mui/material/Tab/tabClasses';
import NotFound from './NotFound';
import { History, LocationState } from 'history';

const styles = (theme: CustomTheme) =>
  createStyles({
    wrapper: {
      fontFamily: theme.typography.fontFamily,
      paddingTop: 64,
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
    },
    tabsIndicator: {
      borderBottom: '8px solid #74ae43',
    },
    tabsRoot: {
      borderBottom: `2px solid ${theme.palette.terra.green}`,
      boxShadow: '0 2px 5px 0 rgba(0,0,0,0.26), 0 2px 10px 0 rgba(0,0,0,0.16)',
      color: '#333F52',
      fontFamily: theme.typography.fontFamily,
      height: 18,
      fontSize: 14,
      fontWeight: 600,
      lineHeight: 18,
      textAlign: 'center',
      width: '100%',
      transition: '0.3s background-color ease-in-out',
    },
    tabSelected: {
      transition: '0.3s background-color ease-in-out',
      backgroundColor: '#ddebd0',
      color: theme.palette.secondary.dark,
      fontWeight: '700 !important',
    },
    component: {
      overflow: 'auto',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
    },
    tabWrapper: {
      display: 'flex',
      zIndex: 11,
    },
    helpIconDiv: {
      borderBottom: `2px solid ${theme.palette.terra.green}`,
      boxShadow: '0 2px 5px 0 rgba(0,0,0,0.26), 0 2px 10px 0 rgba(0,0,0,0.16)',
      float: 'right',
      width: '20px',
    },
  });

interface ITabConfig {
  label: string;
  path: string;
}
const tabConfigs: Array<ITabConfig> = [
  { label: 'Datasets', path: '/datasets' },
  { label: 'Snapshots', path: '/snapshots' },
  { label: 'Activity', path: '/activity' },
  { label: 'Ingest Data', path: '/ingestdata' },
];

interface IRoute {
  path: string;
  component: any;
}
const routes: Array<IRoute> = [
  { path: '/datasets', component: HomeView },
  { path: '/snapshots', component: HomeView },
  { path: '/datasets/new', component: DatasetSchemaCreationView },
  { path: '/datasets/:uuid', component: DatasetOverview },
  { path: '/datasets/:uuid/data', component: DatasetDataView },
  { path: '/snapshots/:uuid', component: SnapshotOverview },
  { path: '/snapshots/:uuid/data', component: SnapshotDataView },
  { path: '/activity', component: HomeView },
];

function Private({ classes }: WithStyles<typeof styles>) {
  const locationSplit = history.location.pathname.split('/');
  const selectedTab = `/${locationSplit[1] || 'datasets'}`;
  return (
    <ConnectedRouter history={history as History<LocationState>}>
      <Router history={history}>
        <div className={classes.wrapper}>
          <Route
            path="/"
            render={() => (
              <Fragment>
                <div className={classes.tabWrapper}>
                  <Tabs
                    value={selectedTab}
                    classes={
                      {
                        root: classes.tabsRoot,
                        indicator: classes.tabsIndicator,
                      } as Partial<TabClasses>
                    }
                  >
                    {tabConfigs
                      .filter(
                        (config: ITabConfig) =>
                          routes.findIndex((route: IRoute) => route.path === config.path) !== -1,
                      )
                      .map((config: ITabConfig, i: number) => (
                        <Tab
                          key={`navbar-link-${i}`}
                          label={config.label}
                          component={Link}
                          value={config.path}
                          to={config.path}
                          classes={{ selected: classes.tabSelected } as Partial<TabClasses>}
                          disableFocusRipple
                          disableRipple
                        />
                      ))}
                  </Tabs>
                  <HelpContainer className={classes.helpIconDiv} />
                </div>
                <div className={classes.component}>
                  <Switch>
                    <Route exact path="/" component={HomeView}>
                      <Redirect to="/datasets" />
                    </Route>
                    {routes.map((route: IRoute, i: number) => (
                      <Route
                        exact
                        key={`route-${i}`}
                        path={route.path}
                        component={route.component}
                      />
                    ))}
                    <Route component={NotFound} />
                  </Switch>
                </div>
              </Fragment>
            )}
          />
        </div>
      </Router>
    </ConnectedRouter>
  );
}

export default withStyles(styles)(Private);
