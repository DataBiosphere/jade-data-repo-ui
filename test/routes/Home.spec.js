import React from 'react';

import { Home } from 'routes/Home';

const props = {
  location: {},
  user: {},
};

function setup(ownProps = props) {
  return mount(<Home {...ownProps} />);
}

describe('Home', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
