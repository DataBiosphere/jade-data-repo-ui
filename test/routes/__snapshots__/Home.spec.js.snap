// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `
.c0 {
  background: #000 linear-gradient(to bottom,#fff 0%,#fff 80%) fixed;
  color: #000;
  min-height: 100vh;
  overflow: hidden;
  position: relative;
}

.c0:before {
  background: linear-gradient(to bottom,#000,#fff);
  bottom: 0;
  content: '';
  left: 0;
  opacity: 0.4;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transform: rotate(-20deg) scale(1.5) translate(0,43%);
  -ms-transform: rotate(-20deg) scale(1.5) translate(0,43%);
  transform: rotate(-20deg) scale(1.5) translate(0,43%);
}

.c1 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

<Background
  data-testid="home"
  key="home"
>
  <div
    className="c0"
    data-testid="home"
  >
    <HomeContainer
      verticalPadding={false}
    >
      <div
        className="c1"
      >
        <div>
          <h1>
            Welcome to the Jade Data Repository
          </h1>
          <div>
            <button
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(66, 133, 244)",
                  "border": "1px solid transparent",
                  "borderRadius": 2,
                  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
                  "color": "#fff",
                  "display": "inline-flex",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "opacity": 0.6,
                  "padding": 0,
                }
              }
              type="button"
            >
              <div
                style={
                  Object {
                    "background": "#fff",
                    "borderRadius": 2,
                    "marginRight": 10,
                    "padding": 10,
                  }
                }
              >
                <svg
                  height="18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#000"
                    fillRule="evenodd"
                  >
                    <path
                      d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                      fill="#EA4335"
                    />
                    <path
                      d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                      fill="#4285F4"
                    />
                    <path
                      d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                      fill="#FBBC05"
                    />
                    <path
                      d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                      fill="#34A853"
                    />
                    <path
                      d="M0 0h18v18H0z"
                      fill="none"
                    />
                  </g>
                </svg>
              </div>
              <span
                style={
                  Object {
                    "fontWeight": 500,
                    "paddingBottom": 10,
                    "paddingLeft": 0,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                Sign in with Google
              </span>
            </button>
          </div>
          <h3>
            New User?
          </h3>
          <h3>
            Jade Data Repository requires a  Google account.
          </h3>
          <p>
            Terra uses your Google account. Once you have signed in and completed the user profile registration step, you can start using Terra.
          </p>
          <a>
            Need to create a Terra account?
          </a>
          <hr />
          <div>
            <div>
              <div>
                WARNING NOTICE
              </div>
              <div>
                You are accessing a US Government web site which may contain information that must be protected under the US Privacy Act or other sensitive information and is intended for Government authorized use only. Unauthorized attempts to upload information, change information, or use of this web site may result in disciplinary action, civil, and/or criminal penalties. Unauthorized users of this website should have no expectation of privacy regarding any communications or data processed by this website. Anyone accessing this website expressly consents to monitoring of their actions and all communications or data transiting or stored on related to this website and is advised that if such monitoring reveals possible evidence of criminal activity, NIH may provide that evidence to law enforcement officials. WARNING NOTICE (when accessing TCGA controlled data) You are reminded that when accessing TCGA controlled access information you are bound by the dbGaP TCGA DATA USE CERTIFICATION AGREEMENT (DUCA).
              </div>
            </div>
          </div>
        </div>
      </div>
    </HomeContainer>
  </div>
</Background>
`;
